build.rs

```diff
@@ -0,0 +1,35 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::process::Command;
+
+// this build script is called on every build of firecracker,
+// after making a change in the source files,
+// embedding the FIRECRACKER_VERSION directly in the resulting binary
+fn main() {
+    let git_version_command = Command::new("git").args(&["describe", "--dirty"]).output();
+
+    match git_version_command {
+        Ok(outcome) => {
+            if outcome.status.success() {
+                println!(
+                    "cargo:rustc-env=FIRECRACKER_VERSION={}",
+                    String::from_utf8(outcome.stdout)
+                        .unwrap()
+                        .trim_start_matches('v')
+                );
+            } else {
+                // if this is not a git repo, return the standard cargo package version
+
+                println!(
+                    "cargo:rustc-env=FIRECRACKER_VERSION={}",
+                    env!("CARGO_PKG_VERSION")
+                );
```

Comment:
This code block is duplicated. Also `"cargo:rustc-env=FIRECRACKER_VERSION={}"` is duplicated a couple of times.

How about something like:

```
    let firecracker_version = Command::new("git")
        .args(&["describe", "--dirty"])
        .output()
        .ok()
        .and_then(|output| {
            if output.status.success() {
                return Some(output.stdout);
            }
            None
        })
        .and_then(|version_bytes| {
            String::from_utf8(version_bytes).ok()
        })
        .and_then(|version_string| {
            Some(version_string.trim_start_matches('v').to_string())
        })
        .unwrap_or(env!("CARGO_PKG_VERSION").to_string());

    println!("cargo:rustc-env=FIRECRACKER_VERSION={}", firecracker_version);
```

If I understand correctly, this is run at build time so performance is not extremely important, but we can also play with `String` and `&str` a bit to optimize it.

Reply:
done